backend/app/api/v1/tasks.py

```
@@ -0,0 +1,453 @@
+    
+    predecessors = task_dependency.get_predecessors(db, task_id=task_id)
+    successors = task_dependency.get_successors(db, task_id=task_id)
+    
+    return predecessors + successors
+
+
+# Task Assignments
+@router.post("/{task_id}/assignments", response_model=TaskAssignment)
+def assign_task(
+    *,
+    db: Session = Depends(get_db),
```

Ruff (0.11.9)

Do not perform function call `Depends` in argument defaults; instead, perform the call within the function, or read the default from a module-level singleton variable

(B008)